hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/CyclicWriteBehindQueue.java

```diff
@@ -50,11 +51,11 @@
      *
      * @see WriteBehindStore#loadable(com.hazelcast.nio.serialization.Data)
      */
-    private final Map<Data, Integer> index;
+    private final Map<Data, MutableInteger> index;
 
     public CyclicWriteBehindQueue() {
         this.deque = new ArrayDeque<DelayedEntry>();
-        this.index = new HashMap<Data, Integer>();
+        this.index = new HashMap<Data, MutableInteger>();
```

Comment:
Why don't you call MultableInteger counter? Makes the code a lot easier to read because the class name now explains its purpose.